### PR TITLE
Analytical_Engine: Add support for all methods required by IElements for analytical interfaces

### LINKED ISSUE #1932: Analytical_Engine: Make add methods needed for IElements to analytiocal element interfaces
#### Description:

related to https://github.com/BHoM/BHoM/issues/962

Add methods as needed by https://github.com/BHoM/documentation/wiki/IElement-required-extension-methods

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### NOTE: Depends on 

Not depending on to compile, but done together with:

https://github.com/BHoM/BHoM/pull/964
   
### Issues addressed by this PR

Closes #1932

- Adding all methods required by IElementXD interfaces as stated https://github.com/BHoM/documentation/wiki/IElement-required-extension-methods to all the analytical interfaces that implements them.

After this, discipline engines can be cleaned up a bit removing the duplicate methods, for cases where the default is enough.

- Adding `Geometry()` query methods to all analytical interfaces.

As the methods are using generic interfaces, the signatures will be slightly more complicated. Also meant I had to resort to things like `Activator.CreateInstance` to generate a new object.

Once these first methods have been added we should have a look into if it would be possible to expand this to more methods across Structure and Env engine.

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EnVkEO1r5zRHmkoQ0gyj8cMBRpXqAOCRxboAB-P45OFF_w?e=L08pBy

### CODE REVIEW

File: Analytical_Engine/Modify/SetElements0D.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        [Input("link", "The ILink to set the IElement0Ds to.")]
+        [Input("newElements0D", "The new IElement0Ds of the ILink. Should be a list of length two, containing exactly two analytical Nodes mathcing the type of the ILink or Geometrical Points. \n" +
+                                "Points will assigin default end properties to the ILink.")]
+        [Output("link", "The ILink with updated Nodes.")]
+        public static ILink<TNode> SetElements0D<TNode>(this ILink<TNode> link, List<IElement0D> newElements0D)
+            where TNode : INode
+        {
+            if (newElements0D.Count != 2)
+            {
+                Reflection.Compute.RecordError("A ILink is defined by 2 nodes.");
```
Comment (FraserGreenroyd) [inline Analytical_Engine/Modify/SetElements0D.cs:51]: Should probably be `An ILink`?

Comment (IsakNaslundBh) [reply Analytical_Engine/Modify/SetElements0D.cs:51]: Yes

**Comment by IsakNaslundBh** [comment]
merged master into current to try to please the CI

**Pull request merged**